DhsTranslator/src/DhsService.h

```diff
@@ -0,0 +1,335 @@
+     * it puts returns false.
+     */
+    template<typename T> bool parseArrayNode(const axutil_env_t *env,
+            axiom_node_t *node, std::vector<T> &array,
+            const std::string &type) {
+        bool ret = true;
+
+        if (axiom_node_get_node_type(node, env) == AXIOM_ELEMENT) {
+            axiom_element_t *nodeElement =
+                    (axiom_element_t *) axiom_node_get_data_element(node, env);
+            axutil_qname_t *typeAttrName = axutil_qname_create(env, "arrayType",
+                    "http://schemas.xmlsoap.org/soap/encoding/", "enc");
+            axiom_attribute_t *typeAttribute = axiom_element_get_attribute(
+                    nodeElement, env, typeAttrName);
+            if (typeAttribute != 0) {
+                axutil_string_t * arrayType = axiom_attribute_get_value_str(
+                        typeAttribute, env);
+                if (std::string(axutil_string_get_buffer(arrayType, env)).compare(
+                        0, type.size(), type) == 0) {
+                    for (axiom_node_t *itemNode = axiom_node_get_first_child(
+                            node, env); itemNode != NULL; itemNode =
+                            axiom_node_get_next_sibling(itemNode, env)) {
+                        if (axiom_node_get_node_type(itemNode, env)
+                                == AXIOM_ELEMENT) {
+                            axiom_element_t *itemElement =
+                                    (axiom_element_t *) axiom_node_get_data_element(
+                                            itemNode, env);
+                            axis2_char_t *itemElementName =
+                                    axiom_element_get_localname(itemElement,
+                                            env);
+                            if (std::string("item").compare(itemElementName)
+                                    == 0) {
+                                axiom_node_t *itemValNode =
+                                        axiom_node_get_first_child(itemNode,
+                                                env);
+                                boost::optional<T> val = parseVal<T>(env,
+                                        itemValNode);
+                                if (val) {
+                                    array.push_back(val.get());
+                                } else {
+                                    ret = false;
+                                }
+                            } else {
+                                ret = false;
+                            }
+                        } else {
+                            ret = false;
+                        }
+                    }
+                } else {
+                    ret = false;
+                }
+            } else {
+                ret = false;
+            }
+            axutil_qname_free(typeAttrName, env);
+        } else {
+            ret = false;
+        }
+        return ret;
+    }
```

Comment:
I think this method can be refactored such as it does not have like 6 levels of nested stuff. It is very hard to read it this way.